src/noisepy/seis/application_modules/esyn_utils.py

```diff
@@ -0,0 +1,122 @@
+import math
+from typing import Tuple
+
+import numpy as np
+import pyasdf
+
+### -----
+# These scripts are aim to perform the 2-D radiative transfer equation
+# for scalar waves (Shang and Gao 1988; Sato 1993),
+# assuming of isotropic scattering and source radiation in infinite medium
+# to calculate synthesized energy densities Esyn.
+### -----
+
+
+### -----
+def read_pyasdf(sfile: str, ccomp: str) -> Tuple[float, float, np.ndarray, np.ndarray]:
+    # useful parameters from each asdf file
+    with pyasdf.ASDFDataSet(sfile, mode="r") as ds:
+        alist = ds.auxiliary_data.list()
+        try:
+            dt = ds.auxiliary_data[alist[0]][ccomp].parameters["dt"]
+            dist = ds.auxiliary_data[alist[0]][ccomp].parameters["dist"]
+            print("working on %s (comp: %s) that is %5.2fkm apart. dt: %.3f " % (sfile, ccomp, dist, dt))
+            # read stacked data
+            sdata = ds.auxiliary_data[alist[0]][ccomp].data[:]
+
+            # time domain variables
+            npts = sdata.size
+            tvec = np.arange(-npts // 2 + 1, npts // 2 + 1) * dt
+            return dist, dt, tvec, sdata
+
+        except Exception:
+            print("continue! no %s component exist" % ccomp)
```

Comment:
```suggestion
            logger.warning(f"continue! no {ccomp} component exist")
```
or maybe `logger.error`?
I also prefer the string interpolation style: https://www.programiz.com/python-programming/string-interpolation. That's more of a preference but is also what we've been using in the rest of the code.

Reply:
It may have other components to be used, so maybe `logger.warning` here should be fine(?).
Good advice on using string interpolation style. That makes it look straightforward. Thanks!